builder/remotecontext/detect.go

```diff
@@ -0,0 +1,172 @@
+package remotecontext
```

Comment:
I did not rename `LazyContext/TarsumContext/modifiableContext` inside this pkg for now. Only in the builder where the conflict with `context.Context` was very clear. More renames could probably be follow up as this is already quite big.